lib/mm_network.pm

```diff
@@ -228,7 +228,7 @@ sub setup_static_mm_network {
     configure_static_ip(ip => $ip, is_nm => $is_nm);
     configure_default_gateway(is_nm => $is_nm);
     configure_static_dns(get_host_resolv_conf(), is_nm => $is_nm);
-    restart_networking(is_nm => $is_nm);
+    restart_networking(is_nm => $is_nm) unless $is_nm;
```

Comment:
Makes sense to me but I would put your git commit message details as a comment in here as some people might be confused why we call `restart_networking` and even tell if it we run with NetworkManager but then actually not call it when we have NetworkManager :)

Reply:
Not sure I understand your last comment but I read it like: "The function promises to restart NM which could confuse people" which I don't think is true. But I can add a comment explaining the reason nevertheless of course.

Reply:
I think @okurz is pondering the same question that came up in my head. This is only used when we don't have network manager.

Reply:
I've added a rather lengthy comment explaining *why* we skip it and how it can be expanded and "properly" implemented if complexity expands in the future (e.g. we support more network management services)

Reply:
It would actually make sense to remove the `is_nm` parameter and to delete all branches within `restart_networking` that would be invoked in case `is_nm` is true. Otherwise this change leaves dead code. (Or is `restart_networking` invoked elsewhere with `is_nm => 1`?)

Reply:
@Martchus according to my understanding of the `git grep` results it looks like `restart_networking(is_nm => 1)` is called elsewhere